spec/services/candidates/registrations/contact_information_spec.rb

```diff
@@ -98,5 +98,45 @@
         end
       end
     end
+
+    context 'postcode is present' do
+      VALID_POSTCODES = [
+        'DN55 1PT',
+        'CR2 6XH',
+        'B33 8TH',
+        'M1 1AE',
+        'W1A 0AX',
+        'EC1A 1BB',
+        'ch63 7ns'
+      ].freeze
+
+      INVALID_POSTCODES = [
+        'horses',
+        'N0T AP057C0D3',
+        'B3333 1BB'
```

Comment:
We should be checking the postcode isn't blank somehow - maybe blank, and just whitespace, within the invalid postcodes list?

Reply:
Theres a seperate context for testing blank postcodes just below this as we show a different error message for blank postcodes and invalid ones